### PR TITLE
Add group attributes to documentation

### PR DESCRIPTION
**Pull request by mekaneck**
Added missing documentation for the attributes of group entities.

## Proposed change

The documentation for group entities did not cover what attributes were populated for a group entity. This fixes that. 

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Enhanced documentation with new sections detailing attributes of new-style and old-style groups.
	- Added tables outlining key attributes for better comprehension of group behavior.

- **Documentation**
	- Clarified differences in behavior between group types, aiding users in managing configurations effectively.

### CODE REVIEW

**Comment by mekaneck** [comment]
This PR documents this behavior: https://github.com/home-assistant/core/issues/77511

File: source/_integrations/group.markdown [hunk 1]
```diff
@@ -155,6 +155,14 @@ To add or remove entities from an existing group, click on `Group options`, all
 
 ![Group members](/images/integrations/group/Group_members.png)
 
+### Group attributes
+
+These are the attributes available for a group. Note that when the group's members are all `unavailable` the group itself becomes `unavailable` and all attributes are removed.
```
Comment (frenck) [inline source/_integrations/group.markdown]: The latter seems a bug? As in, why would we remove the `entity_id` attribute when the group becomes unavailable?

Comment (mekaneck) [reply source/_integrations/group.markdown]: I agree it is undesirable, but my understanding is that every entity (group or any other domain) loses all their attributes when they become unavailable. The desired change would therefore be to make the entity go `unknown` instead. But I expect that to be a feature request and not an issue. There is a current issue on this (https://github.com/home-assistant/core/issues/124055) but previous ones have gone stale. Hence my expectation it needs to be a FR. And therefore we should document the current behavior until this is completed.

Comment (frenck) [reply source/_integrations/group.markdown]: > And therefore we should document the current behavior until this is completed.

I think it is a bug, we shouldn't be documenting bugs. We should fix them.

Comment (mekaneck) [reply source/_integrations/group.markdown]: I've removed the comments related to this behavior.

**Review by frenck — APPROVED** [review]
Thanks, @mekaneck 👍 

../Frenck

**Pull request merged**